### DISCUSSION TITLE
What is the default cache policy of fetch in router handlers?

### DISCUSSION THREAD

**Discussion by nicevlhjfihtqngkldl** [vercel/next.js#68738] [Help]
### Summary

fetch external system's api in router handlers will cache the res by default?
 ```
const data = await fetch('https://api.example.com/...').then((res) =>
    res.json()
  )
 ```

**Comment by Juneezee** [comment]
Yes. Next.js 14 caches by default. You can read more at here: https://nextjs.org/docs/app/api-reference/functions/fetch#optionscache

However, be aware that this is going to change in Next.js 15, see https://nextjs.org/blog/next-15-rc#caching-updates,

**Reply by nicevlhjfihtqngkldl** [reply]
> Yes. Next.js 14 caches by default. You can read more at here: https://nextjs.org/docs/app/api-reference/functions/fetch#optionscache
> 
> However, be aware that this is going to change in Next.js 15, see https://nextjs.org/blog/next-15-rc#caching-updates,

but in the docs, it said: 
"By default, fetch requests retrieve fresh data. Using it will cause a whole route to be [dynamically rendered](https://nextjs.org/docs/app/building-your-application/rendering/server-components#dynamic-rendering) and data will not be cached."

And i do some tests, i use fetch to request multiple times, it always  retrieves the fresh data. So i think the doc is right.

https://nextjs.org/docs/app/building-your-application/data-fetching/fetching
![image](https://github.com/user-attachments/assets/5584433e-ff72-42a6-8ffa-3b1d1470fe5b)

**Reply by Juneezee** [reply]
I think the docs just got updated to Next.js 15

![image](https://github.com/user-attachments/assets/e27afa8f-4484-4ff2-8c72-e93d935a10a4)

**Reply by Juneezee** [reply]
You can read the version 14 docs here: https://github.com/vercel/next.js/blob/v14.2.5/docs/02-app/01-building-your-application/02-data-fetching/01-fetching-caching-and-revalidating.mdx#caching-data

> By default, Next.js automatically caches the returned values of fetch in the [Data Cache](https://github.com/vercel/next.js/blob/v14.2.5/docs/app/building-your-application/caching#data-cache) on the server. This means that the data can be fetched at build time or request time, cached, and reused on each data request.

**Reply by Juneezee** [reply]
> And i do some tests, i use fetch to request multiple times, it always retrieves the fresh data. So i think the doc is right.

Which version of Next.js are you using?

**Reply by nicevlhjfihtqngkldl** [reply]
> > And i do some tests, i use fetch to request multiple times, it always retrieves the fresh data. So i think the doc is right.
> 
> Which version of Next.js are you using?

Yes,you are right, i just found that i was using next.js 15. 
Thanks for your help!
```
"dependencies": {
    "next": "15.0.0-rc.0",
  },
```

**Reply by xingpingcn** [reply]
does it mean that https://nextjs.org/docs include the two versions of nextjs?

in [caching](https://nextjs.org/docs/app/building-your-application/caching) 

> By default, data requests that use fetch are not cached. You can use the [cache](https://nextjs.org/docs/app/building-your-application/caching#fetch-optionscache) and [next.revalidate](https://nextjs.org/docs/app/building-your-application/caching#fetch-optionsnextrevalidate) options of fetch to configure the caching behavior.

in [fetching](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching)

> By default, fetch requests retrieve fresh data. Using it will cause a whole route to be [dynamically rendered](https://nextjs.org/docs/app/building-your-application/rendering/server-components#dynamic-rendering) and data will not be cached.

also in [fetching](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching)

> Good to know: In Next.js 14 and earlier, fetch requests were cached by default. See the [upgrade guide](https://nextjs.org/docs/app/building-your-application/upgrading/version-15) for more information.

and which one is right? if I'm not mistaken, nextjs 14 fetch API will cache by default, nextjs 15 fetch API will NOT cache by default. the docs is confusing me.

**Reply by xingpingcn** [reply]
and btw, how can i change the version of docs from 15 to 14? i can not find an option to shift on official docs.
